tilewe/engine.py

```diff
@@ -139,11 +139,12 @@ def on_search(self, board: tilewe.Board, _seconds: float) -> tilewe.Move:
         random.shuffle(moves) 
         
         player = board.current_player
+        N = board.n_players
 
         def eval_after_move(m: tilewe.Move) -> int: 
             with MoveExecutor(board, m):
                 total = 0
-                for color in range(board.n_players): 
+                for color in range(N): 
```

Comment:
only accesses the C API for this value once per ply instead of once per move per ply 🤷 not sure if really matters or not, but is a no-op even if it doesn't matter